Tests/SwiftProtobufTests/Test_TextFormat_Map_proto3.swift

```diff
@@ -47,11 +47,82 @@ class Test_TextFormat_Map_proto3: XCTestCase, PBTestHelpers {
         assertTextFormatDecodeFails("map_int32_int32 {key:1}")
     }
 
+    func test_Int32Int32_numbers() {
+        assertTextFormatDecodeSucceeds("1 {\n  key: 1\n  value: 2\n}\n") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {key: 1, value: 2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {key: 1; value: 2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {key:1 value:2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {key:1 value:2}\n1 {key:3 value:4}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
+        assertTextFormatDecodeSucceeds("1 [{key:1 value:2}, {key:3 value:4}]") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
+        assertTextFormatDecodeSucceeds("1 [{key:1 value:2}];1 {key:3 value:4}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
+        assertTextFormatDecodeFails("1 [{key:1 value:2},]")
+        assertTextFormatDecodeFails("1 [{key:1 value:2}")
+        assertTextFormatDecodeFails("1 [{key:1 value:2 nonsense:3}")
+        assertTextFormatDecodeFails("1 {key:1}")
+
+        // Using numbers for "key" and "value" in the map entries.
+
+        assertTextFormatDecodeSucceeds("1 {\n  1: 1\n  2: 2\n}\n") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {1: 1, 2: 2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {1: 1; 2: 2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {1:1 2:2}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2]
+        }
+        assertTextFormatDecodeSucceeds("1 {1:1 2:2}\n1 {1:3 2:4}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
+        assertTextFormatDecodeSucceeds("1 [{1:1 2:2}, {1:3 2:4}]") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
+        assertTextFormatDecodeSucceeds("1 [{1:1 2:2}];1 {1:3 2:4}") {(o: MessageTestType) in
+            return o.mapInt32Int32 == [1:2, 3:4]
+        }
```

Comment:
Should we include a case with `,` and `;` separators within the map elements?  E.g.
```
     1 [{1:1,2:2},{1:3;2:4}]
```